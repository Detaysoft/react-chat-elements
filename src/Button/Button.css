.rce-button {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    font-size: 12px;
    border-radius: 5px;
    cursor: pointer;
    padding: 10px;
    text-align: center;
    box-sizing: border-box;
    background: #3979aa;
    color: white;
    transition: all 0.15s ease;
    user-select: none;
    border: none;
    outline: none;
    border: none;
    position: relative;
}

.rce-button-icon--container {
    display: flex;
    align-items: center;
}

.rce-button:hover {
    opacity: 0.8;
}

.rce-button:active {
    opacity: 0.6;
}

.rce-button.outline {
    background: rgba(0, 0, 0, 0) !important;
    border: 1px solid #3979aa;
    color: #3979aa;
}

.rce-button.outline:hover {
    opacity: 0.6;
}

.rce-button.outline:active {
    opacity: 0.3;
}

.rce-button.transparent {
    background: rgba(0, 0, 0, 0) !important;
}

.rce-button.transparent:hover {
    opacity: 0.6;
}

.rce-button.transparent:active {
    opacity: 0.3;
}

.rce-button-icon {
    position: relative;
    font-size: 18px;
    display: flex;
    padding: 0 3px;
}

.rce-button-badge{
    border-radius: 4px;
    padding: 4px;
    background: #f64b34;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    position: absolute;
    right: -7px;
    top:-7px;
    font-size: 10px;
}