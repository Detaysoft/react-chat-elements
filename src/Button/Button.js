import React, { Component } from 'react';
import './Button.css';

const classNames = require('classnames');

export class Button extends Component {
    render() {
        return (
            <button
                ref={this.props.buttonRef}
                title={this.props.title}
                className={classNames('rce-button', this.props.type, this.props.className)}
                style={{
                    backgroundColor: this.props.backgroundColor,
                    color: this.props.color,
                    borderColor: this.props.backgroundColor
                }}
                disabled={this.props.disabled}
                onClick={this.props.onClick}>
                {
                    this.props.icon ?
                        <span className='rce-button-icon--container'>
                            {(this.props.icon.float === 'right' || !this.props.icon.float) && <span>{this.props.text}</span>}

                            <span style={{ float: this.props.icon.float, fontSize: this.props.icon.size || 12 }} className='rce-button-icon'>{this.props.icon.component}</span>

                            {this.props.icon.float === 'left' && <span>{this.props.text}</span>}
                        </span>
                        : <span>{this.props.text}</span>
                }
            </button>
        );
    }
}

Button.defaultProps = {
    text: '',
    disabled: false,
    type: null,
    icon: null,
    backgroundColor: '#3979aa',
    color: 'white',
    className: null,
    buttonRef: null,
    title: null,
};

export default Button;
