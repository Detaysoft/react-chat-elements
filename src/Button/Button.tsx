import './Button.css'
import classNames from 'classnames'
import { IButtonProps } from '../type'

const Button: React.FC<IButtonProps> = ({ disabled = false, backgroundColor = '#3979aa', color = 'white', ...props }) => {
  return (
    <button
      ref={props.buttonRef}
      title={props.title}
      className={classNames('rce-button' as any, props.type as any, props.className as any)}
      style={{
        backgroundColor: backgroundColor,
        color: color,
        borderColor: backgroundColor,
      }}
      disabled={disabled}
      onClick={props.onClick}
    >
      {props.icon ? (
        <span className='rce-button-icon--container'>
          {(props.icon.float === 'right' || !props.icon.float) && <span>{props.text}</span>}

          <span style={{ float: props.icon.float, fontSize: props.icon.size || 12 }} className='rce-button-icon'>
            {props.icon.component}
          </span>

          {props.icon.float === 'left' && <span>{props.text}</span>}
        </span>
      ) : (
        <span>{props.text}</span>
      )}
    </button>
  )
}
export default Button
