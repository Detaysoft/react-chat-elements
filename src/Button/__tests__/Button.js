import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../Button';

describe('Button component', () => {
  it('should render without issues', () => {
    const component = shallow(<Button />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});