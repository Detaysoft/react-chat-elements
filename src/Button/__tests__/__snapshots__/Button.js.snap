// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render without issues 1`] = `
<button
  className="rce-button"
  disabled={false}
  style={
    Object {
      "backgroundColor": "#3979aa",
      "borderColor": "#3979aa",
      "color": "white",
    }
  }
>
  <span />
</button>
`;
